### PR TITLE
Compound liquidations

### PR DESCRIPTION
**Pull request by nebolax**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/chain/ethereum/modules/compound/decoder.py [hunk 1]
```diff
@@ -207,6 +216,78 @@ def _decode_borrow_and_repay(
...
+    def _decode_liquidate(
+            self,
+            transaction_hash: 'EVMTxHash',
+            tx_log: 'EvmTxReceiptLog',
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> DecodingOutput:
+        """Decode a liquidation event happening over a tracked account"""
+        borrower = hex_or_bytes_to_address(tx_log.data[32:64])
+        if self.base.is_tracked(borrower) is False:
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/compound/decoder.py:228]: what about the liquidator's side? What if he is using rotki?

File: rotkehlchen/chain/ethereum/modules/compound/decoder.py [hunk 2]
```diff
@@ -207,6 +216,78 @@ def _decode_borrow_and_repay(
...
+            chain_id=self.base.evm_inquirer.chain_id,
+            token_kind=EvmTokenKind.ERC20,
+            evm_inquirer=self.base.evm_inquirer,
+            protocol=CPT_COMPOUND,
+        )
+        seized_collateral_amount = asset_normalized_value(
+            amount=seize_amount_raw,
+            asset=collateral_ctoken,
+        )
+        # use the logs to know what token was getting repayed for the position. We have the ctoken
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/compound/decoder.py:248]:
```suggestion
        # use the logs to know what token was getting repaid for the position. We have the ctoken
```

File: rotkehlchen/chain/ethereum/modules/compound/decoder.py [hunk 3]
```diff
@@ -207,6 +216,78 @@ def _decode_borrow_and_repay(
...
+                    token_kind=EvmTokenKind.ERC20,
+                    evm_inquirer=self.base.evm_inquirer,
+                )
+                break
+
+        if repaying_token is None:
+            log.error(f'Failed to decode compound liquidation at {transaction_hash.hex()}')
+            return DEFAULT_DECODING_OUTPUT
+
+        repayed_amount = asset_normalized_value(
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/compound/decoder.py:271]:
```suggestion
        repaid_amount = asset_normalized_value(
```

File: rotkehlchen/chain/ethereum/modules/compound/decoder.py [hunk 4]
```diff
@@ -207,6 +216,78 @@ def _decode_borrow_and_repay(
...
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == borrower and
+                event.address == liquidator_address and
+                event.balance.amount == seized_collateral_amount and
+                event.asset == collateral_ctoken
+            ):
+                event.event_subtype = HistoryEventSubType.LIQUIDATE
+                event.notes = f'Liquidate {seized_collateral_amount} {collateral_ctoken.symbol} to repay {repayed_amount} {repaying_token.symbol} on compound'  # noqa: E501
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/compound/decoder.py:285]: This is from the view of the borrower/victim. So it should be:

Lost {
```suggestion
                event.notes = f'Lost {seized_collateral_amount} {collateral_ctoken.symbol} in a compound forced liquidation to repay {repayed_amount} {repaying_token.symbol}'  # noqa: E501
```

File: rotkehlchen/chain/ethereum/modules/compound/decoder.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/compound/decoder.py:228]: Added it to handle the normal path. For liquidations happening using MEV bots we have to make additional changes to our logic

File: rotkehlchen/chain/ethereum/modules/compound/decoder.py [hunk 5]
```diff
@@ -207,6 +216,94 @@ def _decode_borrow_and_repay(
...
+            elif (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.PAYBACK_DEBT and
+                event.location_label == liquidator_address and
+                event.balance.amount == repaid_amount and
+                event.asset == repaying_token and
+                event.counterparty == CPT_COMPOUND
+            ):
+                event.notes = f'Repay {repaid_amount} {repaying_token.symbol} in a compound liquidation'  # noqa: E501
+                event.extra_data = {'in_liquidation': True}
```
Comment (LefterisJP) [inline rotkehlchen/chain/ethereum/modules/compound/decoder.py:293]: What is this used for?

**Review by LefterisJP — APPROVED** [review]
LGTM

File: rotkehlchen/chain/ethereum/modules/compound/decoder.py (hunk 5 above)
Comment (yabirgb) [reply rotkehlchen/chain/ethereum/modules/compound/decoder.py:293]: To know for the paybacks which one happens due to a liquidation at the moment of computing statistics

**Pull request merged**